src/components/Contracts/ContractAgreementsTable.js

```diff
@@ -0,0 +1,328 @@
+    showInfoIcon: false,
+  },
+];
+
+/**
+ * @function
+ * @name prepareRows
+ * @description transform rows such that are well formated compatible with the table meta data
+ * @param {object} rows - rows to be transformed
+ * @returns {Array} - transformed rows
+ */
+const prepareRows = (rows) =>
+  rows.map((row) => {
+    return {
+      ...row,
+      created_at: convertDateStringToHumanReadableFormat(
+        row.created_at,
+        'yyyy-MM-dd'
+      ),
+      updated_at: convertDateStringToHumanReadableFormat(
+        row.updated_at,
+        'yyyy-MM-dd'
+      ),
+      // csv_start_date: row.consolidation_period_start,
+      // csv_end_date: row.consolidation_period_end,
+      // consolidation_period_start: convertDateStringToHumanReadableFormat(
+      //   row.consolidation_period_start,
+      //   'yyyy-MM-dd'
+      // ),
+      // consolidation_period_end: convertDateStringToHumanReadableFormat(
+      //   row.consolidation_period_end,
+      //   'yyyy-MM-dd'
+      // ),
+      // calculated_at: convertDateStringToHumanReadableFormat(
+      //   row.calculated_at,
+      //   'yyyy-MM-dd'
+      // ),
+      // payment_confirmed_at: convertDateStringToHumanReadableFormat(
+      //   row.payment_confirmed_at
+      // ),
+      // paid_at: row.paid_at ? format(new Date(row.paid_at), 'yyyy-MM-dd') : '',
+    };
+  });
+
+/**
+ * @function
+ * @name ContractAgreementsTable
+ * @description renders the contracts table
+ *
+ * @returns {React.Component} - contracts table component
+ * */
+function ContractAgreementsTable() {
+  const classes = useStyles();
+  // state for contracts table
+  const [contracts, setContracts] = useState([]);
+  const [activeDateRangeString, setActiveDateRangeString] = useState('');
+  const [filter, setFilter] = useState({});
+  const [page, setPage] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [contractsPerPage, setContractsPerPage] = useState(20);
+  const [sortBy, setSortBy] = useState({
+    field: 'status',
+    order: 'desc',
+  });
+  const [isDateFilterOpen, setIsDateFilterOpen] = useState(false);
+  const [isMainFilterOpen, setIsMainFilterOpen] = useState(false);
+  const [totalContracts, setTotalContracts] = useState(0);
+  const [selectedContractAgreement, setSelectedContractAgreement] = useState(
+    null
+  );
+  const [isDetailShown, setDetailShown] = useState(false);
+
+  async function getContracts(fetchAll = false) {
+    // console.warn('getContracts with fetchAll: ', fetchAll);
+    setIsLoading(true); // show loading indicator when fetching data
+
+    const { results, totalCount = 0 } = await getContractAgreements(fetchAll);
+    console.log('results:', results, 'totalCount:', totalCount);
+    setContracts(results);
+    setTotalContracts(totalCount);
+
+    setIsLoading(false); // hide loading indicator when data is fetched
+  }
+
+  async function getContractAgreements(fetchAll = false) {
+    const filtersToSubmit = { ...filter };
```

Comment:
I'd expect the `Name` query to be case-insensitive and support partial matches, but this may be an API issue.